### PR TITLE
reuse test container locally

### PR DESCRIPTION
**Pull request by ckiosidis**
Signed-off-by: Babis Kiosidis <ckiosidis@gmail.com>
# TL;DR

Set the config in your local environment to reuse the flyte sandbox test container throughout the tests. The config can only be set in the home directory

```
 cat ~/.testcontainers.properties 
testcontainers.reuse.enable=true
```

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: integration-tests/src/test/java/org/flyte/utils/FlyteSandboxContainer.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class FlyteSandboxContainer extends GenericContainer<FlyteSandboxContaine
 
   public static final String IMAGE_NAME = "ghcr.io/flyteorg/flyte-sandbox:v1.1.0";
 
-  public static final FlyteSandboxContainer INSTANCE = new FlyteSandboxContainer();
+  public static final FlyteSandboxContainer INSTANCE = new FlyteSandboxContainer().withReuse(true);
```
Comment (narape) [inline integration-tests/src/test/java/org/flyte/utils/FlyteSandboxContainer.java:38]:
```suggestion
  public static final FlyteSandboxContainer INSTANCE = new FlyteSandboxContainer()
          // Note to the developer: to enable test container reuse, please do the following
          // echo testcontainers.reuse.enable=true > ~/.testcontainers.properties
         .withReuse(true);
```

**Review by narape — APPROVED** [review]
I added a suggestion to add a comment as it would be easier to find that this PR description

File: integration-tests/src/test/java/org/flyte/utils/FlyteSandboxContainer.java (hunk 1 above)
Comment (narape) [reply integration-tests/src/test/java/org/flyte/utils/FlyteSandboxContainer.java:38]: Maybe reformatting is needed after applying my sugestion

File: integration-tests/src/test/java/org/flyte/SerializeJavaIT.java [hunk 2]
```diff
@@ -21,21 +21,25 @@
...
 
   @Test
-  @Disabled
   public void testSerializeWorkflows() {
     try {
-      CLIENT.serializeWorkflows(CLASSPATH, tempDir.getAbsolutePath());
+      File current = new File("target/protos");
+      current.mkdir();
+
+      File tempDir = managed.resolve(current.getAbsolutePath()).toFile();
```
Comment (narape) [inline integration-tests/src/test/java/org/flyte/SerializeJavaIT.java:40]: I don't understand why you make `mkdir` first and then resolve. In my head this makes more sense

```java
  @TempDir File tempDir;

//...

    try {
      File protosDir = managed.resolve("target/protos").toFile();
      protosDir.mkdir();
      CLIENT.serializeWorkflows(CLASSPATH, protosDir.getPath());
...

Comment (ckiosidis) [reply integration-tests/src/test/java/org/flyte/SerializeJavaIT.java]: this wont work because the default location for a @TempDir is somewhere in a tmp /var/... and the docker images that we run do not have access to that path.
We need to first create the File that points in the project's path, then we pass it to the tempdir

```
[docker-java-stream--796442191] INFO 🐳 [ghcr.io/flyteorg/flytekit-java:0.3.20-SNAPSHOT] - STDERR: Caused by: java.io.FileNotFoundException: /var/folders/vl/46bdmk_x37s3vd61fcfp05f80000gn/T/junit15098090948395355541/target/protos/00_org.flyte.examples.PrintMessageTask_1.pb (No such file or directory)
```
